akka-http-tests/src/test/scala/akka/http/scaladsl/CustomMediaTypesSpec.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import akka.http.scaladsl.client.RequestBuilding
+import akka.http.scaladsl.model.MediaType.WithFixedCharset
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.server.{ Directives, RoutingSpec }
+import akka.http.scaladsl.settings.{ ParserSettings, ServerSettings }
+import akka.stream.ActorMaterializer
+import akka.testkit.AkkaSpec
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+import scala.concurrent.duration._
+
+class CustomMediaTypesSpec extends AkkaSpec with ScalaFutures
+  with Directives with RequestBuilding {
+
+  implicit val mat = ActorMaterializer()
+
+  "Http" should {
+    "allow registering custom media type" in {
+      import system.dispatcher
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+
+      //#application-custom
+
+      // similarily in Java: `akka.http.javadsl.settings.[...]`
+      import akka.http.scaladsl.settings.ParserSettings
+      import akka.http.scaladsl.settings.ServerSettings
+
+      // define custom media type:
+      val utf8 = HttpCharsets.`UTF-8`
```

Comment:
only in order to make the line below fit in a single line when included into docs